docs/1-tendermint-overview.md

```diff
@@ -0,0 +1,92 @@
+
+### Idea behind the proof
+
+The idea behind the proof is that it can never happen that there are two sets of *2f + 1* Prevote or Precommit messages in the same round *r* for different values. If that would be the case, we'll have at least *f + 1* processes that sent both the messages, which would mean *f + 1* are faulty. But at most *f* processes are faulty, so this is impossible.
+Moreover, at most one value can be locked in a round and if a correct process decided value *v* in round *r*, *v* will be locked for all the next rounds after *r* on that height and correct processes will not send Prevote/Precommit message for values different than *v*. Therefore, it's clear that *v* will be the only value that could be decided from that moment onwards.
+
+### Proof
+
+Let's assume that a correct process p decides value v in round r of height h. We want to prove that any other correct process p' in some round r' >= r of height h decides v' such that v' = v.
+
+We have two cases:
+
+- if r' = r: 
+
+    p' has decided value v' in round r, so it has received at least 2f + 1 Precommit messages for value v' in round r. 
+    p has decided value v in the same round r, so it has received at least 2f + 1 Precommit messages for value v in round r. 
+    As it has been shown previously, any two sets of 2f + 1 messages intersect in at least one correct process. Since a correct process only sends a single Precommit message, it must be that v = v'.  
+
+- if r ' > r:
+        
+    p has decided value v in the same round r, so it has received at least 2f + 1 Precommit messages for value v in round r.
+    Since the number of faulty processes is at most f, at least f+1 correct processes have locked value v by round r and, by algorithm rules, they will send Prevote messages only for value v in all the next rounds of height h.
+    
+    p' has decided value v' in round r' > r, so it has received at least 2f + 1 Precommit messages for value v' in round r' > r. 
+    Since the number of faulty processes is at most f, at least f+1 correct processes have locked value v' by round r' > r and these processes must have received at least 2f + 1 Prevote messages for value v' by round r'.
+    
+    From the fact that the intersection of 2f + 1 processes has at least one correct process, at least one correct process that locked value v in round r also sent a Prevote message for value v' in a later round r' > r. 
+    Since this is impossible, it can only be that v = v'.   
+
+Since these are the two only possible cases, the above reasoning proves that the agreement property is satisfied.
+
+
+In conclusion, when more than 1/3 of processes are faulty, a fork can happen and the agreement property of the consensus algorithm could be violated. Otherwise, the agreement property is always satisfied.
```

Comment:
I would avoid using "fork" since we did not define that. I would simply use "violation of agreement" or "disagreement".

Reply:
Fixed.